test/regexmanager.cpp

```diff
@@ -106,3 +106,33 @@ TEST_CASE("`highlight all` adds rules for all locations", "[RegexManager]")
 		}
 	}
 }
+
+TEST_CASE("RegexManager does not hang on regexes that can match empty strings", "[RegexManager]")
+{
+	RegexManager rxman;
+	std::string input = "The quick brown fox jumps over the lazy dog";
+	std::string compare = input;
```

Comment:
This line is unused, please remove it.

Reply:
Which line is unused?

Reply:
OH nevermind, im an idiot

Reply:
Don't be hard on yourself, the tests are very similar and it's not obvious that you don't need this line. It's unfortunate that compilers can't issue "unused value" warnings for non-POD types.